From cd12d9dd2bea8e4580f458df77229477fc55ba70 Mon Sep 17 00:00:00 2001
From: James Hilliard <james.hilliard1@gmail.com>
Date: Wed, 14 May 2025 14:17:40 -0600
Subject: [PATCH] Update cmake_minimum_required to 3.5

This supports compilation with cmake-4.0.0.

Fixes:
CMake Error at CMakeLists.txt:5 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

Signed-off-by: James Hilliard <james.hilliard1@gmail.com>
Upstream-Status: Backport [cd12d9dd2bea8e4580f458df77229477fc55ba70]
---
 CMakeLists.txt     | 2 +-
 src/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 796d7bc..0a11730 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier:     LGPL-2.1-or-later

-cmake_minimum_required (VERSION 2.6)
+cmake_minimum_required (VERSION 3.5)
 project (libubootenv C)

 # The version number.
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c56d0c7..3370232 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,7 +1,7 @@
 # SPDX-FileCopyrightText: 2019-2021 Stefano Babic <stefano.babic@swupdate.org>
 #
 # SPDX-License-Identifier:     LGPL-2.1-or-later
-cmake_minimum_required (VERSION 2.6)
+cmake_minimum_required (VERSION 3.5)
 # Sources and private headers
 SET(libubootenv_SOURCES
   uboot_env.c
